src/commands/dev/edge/watch.rs

```diff
@@ -38,15 +38,29 @@ pub fn watch_for_changes(
         // this allows the server to route subsequent requests
         // to the proper script
         let uploaded = setup::upload(&mut target, &deploy_target, &user, session_token, verbose);
+
         match uploaded {
             Ok(token) => {
                 *preview_token = token;
             }
             Err(err) => {
+                // TODO: should probably replace legacy client to handle API errors properly.
+                //       Also shutdown `wrangler dev` more gracefully.
                 if let Some(err) = err.downcast_ref::<setup::BadRequestError>() {
-                    // no need to refresh session here, bad request indicates problem with user code
-                    StdOut::warn(&err.0);
+                    // if the API error code is 10049, then it's an expired preview token
+                    if err.0.contains("10049") {
+                        refresh_session_channel.send(Some(()))?;
+                        break;
+                    } else if err.0.contains("10021") {
+                        // Continue to watch for changes if it's a syntax error
+                        StdOut::warn(&format!("{}\nPlease update your code.", &err.0));
```

Comment:
Checking, does err contain the actual syntax error detected, with a line number, and will it get logged?

Reply:
<img width="728" alt="Screen Shot 2021-09-15 at 2 37 30 PM" src="https://user-images.githubusercontent.com/46542932/133513114-992d9afc-364c-4427-bf8d-b28db8369d62.png">

Yes, it shows the syntax error and line number. This is printed to `stdout`, and the `reqwest` library logs the HTTP response received from the API.